rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -201,13 +286,15 @@ __set_parameters_atomically_common(
   if (on_set_parameters_callback) {
     result = on_set_parameters_callback(parameters);
   }
+  result.successful =
+    result.successful && __check_parameters(parameter_infos, parameters);
```

Comment:
It's fine for `__check_parameters` to return a bool; maybe I'd add a reason here in case it doesn't succeed.

Reply:
Thanks! I forgot checking what `rcl_interfaces::msg::SetParametersResult` was.